src/main/java/com/powsybl/openloadflow/ac/outerloop/AreaInterchangeControlOuterloop.java

```diff
@@ -0,0 +1,160 @@
+        context.setData(contextData);
+    }
+
+    @Override
+    public OuterLoopResult check(AcOuterLoopContext context, ReportNode reportNode) {
+        List<LfArea> areas = context.getNetwork().getAreas();
+        double slackBusActivePowerMismatch = context.getLastSolverResult().getSlackBusActivePowerMismatch();
+
+        List<LfArea> areasToBalance = areas.stream()
+                .filter(area -> {
+                    double areaActivePowerMismatch = getInterchangeMismatchWithSlack(area, slackBusActivePowerMismatch);
+                    double absMismatch = Math.abs(areaActivePowerMismatch);
+                    return absMismatch > this.areaInterchangePMaxMismatch / PerUnit.SB && absMismatch > ActivePowerDistribution.P_RESIDUE_EPS;
+                })
+                .toList();
+        boolean shouldBalance = !areasToBalance.isEmpty();
+
+        if (!shouldBalance) {
+            // Balancing takes the slack mismatch of the Areas into account. Now that the balancing is done, we check only the interchange power flow mismatch.
+            // Doing this we make sure that the Areas' interchange targets have been reached and that the slack is correctly distributed.
+            Map<LfArea, Double> interchangeMismatches = areas.stream().filter(area -> {
+                double areaInterchangeMismatch = getInterchangeMismatch(area);
+                double absMismatch = Math.abs(areaInterchangeMismatch);
+                return absMismatch > this.areaInterchangePMaxMismatch / PerUnit.SB && absMismatch > ActivePowerDistribution.P_RESIDUE_EPS;
+            }).collect(Collectors.toMap(area -> area, AreaInterchangeControlOuterloop::getInterchangeMismatch));
+
+            if (interchangeMismatches.isEmpty()) {
+                LOGGER.debug("Already balanced");
+            } else {
+                interchangeMismatches.forEach((area, mismatch) ->
+                        LOGGER.error("Failed area interchange control: Area {} is not balanced, remains {} MW mismatch", area.getId(), mismatch * PerUnit.SB));
+            }
+            return new OuterLoopResult(this, OuterLoopStatus.STABLE);
+        }
+        double totalDistributedActivePower = 0.0;
+        int totalIterations = 0;
+        boolean movedBuses = false;
+        Map<LfArea, Double> remainingMismatchMap = new HashMap<>();
+        for (LfArea area : areasToBalance) {
+            double areaActivePowerMismatch = getInterchangeMismatchWithSlack(area, slackBusActivePowerMismatch);
+            ActivePowerDistribution.Result result = activePowerDistribution.run(area.getBuses(), areaActivePowerMismatch);
+            double remainingMismatch = result.remainingMismatch();
+            double distributedActivePower = areaActivePowerMismatch - remainingMismatch;
+            totalDistributedActivePower += distributedActivePower;
+            movedBuses |= result.movedBuses();
+            totalIterations += result.iteration();
```

Comment:
I don' think this sum has anything meaningful, only the value per area is meaningful.

Reply:
Refactored so that in the logging and reports we see the value per area. 
Now the totalDistributedActivePower remains used only for the  "Ac loadflow complete" log which has "distributedActivePower = ..."
I started to think about how to update this log. Do you have any ideas ?